### PR TITLE
 [DRAFT][luci-interpreter] Introduce Cast

### PR DESCRIPTION
**Pull request by Bronnikoff**
This draft introduces Cast to luci-interpreter.

ONE-DCO-1.0-Signed-off-by: Maksim Bronnikov <max120199@gmail.com>

---------------

For: #7261

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Cast.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      cast_from_pointer_to_tensor(getTensorData<int64_t>(in_tensor), out_tensor);
+      break;
+    case loco::DataType::FLOAT32:
+      cast_from_pointer_to_tensor(getTensorData<float>(in_tensor), out_tensor);
+      break;
+    case loco::DataType::BOOL:
+      cast_from_pointer_to_tensor(getTensorData<bool>(in_tensor), out_tensor);
+      break;
+    default:
+      throw std::runtime_error("Unsupported input type.");
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/Cast.cpp:112]: Did you take this from tflite kernels?
If so, better to add tensorflow copyright in boilerplace.

Comment (binarman) [reply compiler/luci-interpreter/src/kernels/Cast.cpp:112]: Another option is to write translation from loco types to tflite types and reuse tflite code.

Comment (Bronnikoff) [reply compiler/luci-interpreter/src/kernels/Cast.cpp:113]: > Did you take this from tflite kernels?

I used https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/cast.cc for kernel implementation.

> If so, better to add tensorflow copyright in boilerplace.

Done =)

**Comment by Bronnikoff** [comment]
Close, because changes merged and issue closed.

**Pull request closed without merging**